internal/editor/editor.go

```diff
@@ -47,8 +59,17 @@ func Launch(repo *git.Repo, config Config) (string, error) {
 		return "", err
 	}
 
-	// launch the editor as a subprocess
-	cmd := exec.Command(config.Command, tmp.Name())
+	// Launch the editor as a subprocess.
+	// We interpret the command with shell syntax to allow users to specify
+	// both flags and use editor executables with spaces.
+	// e.g., EDITOR="'/path/with spaces/editor'" or
+	// EDITOR="code --wait" work.
+	args, err := shellquote.Split(config.Command)
```

Comment:
So if I understand this correctly, you are using a different interpreter for executing the command? Is that why code --wait didn't work for Nick?

Reply:
`shellquote.Split` just converts a string into arguments. So an input of `"/foo bar/spam" --eggs` gets converted to `[]string{"/foo bar/spam", "--eggs"}`. There's no "interpreting" happening here, it just applies shell string quoting rules to convert the string into a list of arguments (which ultimately forms the `argv` of the command that's executed).

Reply:
yeh I think i inferred that, the question was rather why that was causing `Command` to fail, I'm guessing this is how it works. The first param itself need to be the actual command (without params).